### DISCUSSION TITLE
ffmpeg in Windows builds

### DISCUSSION THREAD

**Discussion by axfelix** [ImageMagick/ImageMagick#2417] [Development] [answered]
Hi,

This is a rather specific edge case, but I've noticed that Imagemagick currently provides very good and up to date hwaccel-enabled builds of ffmpeg on Windows. However, you aren't building with all of the CUDA functions available as documented in https://developer.nvidia.com/ffmpeg:

`./configure --enable-cuda-sdk --enable-cuvid --enable-nvenc --enable-nonfree --enable-libnpp`

And this, notably, makes it impossible to run an ffmpeg conversion job with both GPU *decoding* in addition to encoding, as adding an argument like `-vf "hwupload_cuda,scale_npp=w=1280:h=-2"` is necessary in order to correctly parse video dimensions, per this ffmpeg bug: https://trac.ffmpeg.org/ticket/5587

Is there any chance you can start building ffmpeg with CUDA when you include it on Windows?

Thanks!

**Comment by dlemstra** [comment, marked as answer]
We are using the official binaries and don't build them ourselves. We download them from here: https://ffmpeg.zeranoe.com/builds/.
